auth.php

```diff
@@ -455,14 +455,15 @@ public function saml_login() {
         require('setup.php');
         require_once("$CFG->dirroot/login/lib.php");
 
-        // Set the default IdP to be the first in the list. Used when dual login is disabled.
-        $arr = array_reverse($saml2auth->metadataentities);
-        $metadataentities = array_pop($arr);
-        $idpentity = array_pop($metadataentities);
-        $idp = md5($idpentity->entityid);
+        if (isset($_GET['idp'])) {
```

Comment:
I know this is moved code, but any reason for not using optional_param ?

Reply:
No reason, let's use that instead, have updated that now